### PR TITLE
fix(templates): Include package license metadata conditionally in templates

### PR DESCRIPTION
**Pull request by ReubenFrankel**
When I choose not to include a license, I expect there not to be any metadata pertaining to a particular license type in the `pyproject.toml`.

Unsure how tests work for templates, so haven't included any for now - I am happy to try with some direction.

## Summary by Sourcery

Chores:
- Remove unnecessary license information when a license is not specified.

----
📚 Documentation preview 📚: https://meltano-sdk--2862.org.readthedocs.build/en/2862/

### CODE REVIEW

**Comment by ReubenFrankel** [comment]
This AI review of the PR is crazy :open_mouth:

**Comment by ReubenFrankel** [comment]
Let me know how you feel about https://github.com/meltano/sdk/pull/2862/commits/7b679287f41a7fc6f8260a4da5c88acc94842f96.

**Comment by edgarrmondragon** [comment]
> This AI review of the PR is crazy 😮

Crazy as "ok, this is actually somewhat useful" or "what? that is nonsense!"? 😅

**Comment by edgarrmondragon** [comment]
> Let me know how you feel about [7b67928](https://github.com/meltano/sdk/commit/7b679287f41a7fc6f8260a4da5c88acc94842f96).

@ReubenFrankel was the metadata validation failing before that commit?

File: cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml [hunk 1]
```diff
@@ -18,15 +18,19 @@ keywords = [
...
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
     "Programming Language :: Python :: 3.13",
 ]
-license = "Apache-2.0"
+{%- if cookiecutter.license != "None" %}
+license = "{{cookiecutter.license}}"
+{%- endif %}
 license-files = [ "LICENSE" ]
```
Comment (edgarrmondragon) [inline cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml:34]: (let's start a thread)

> Let me know how you feel about [7b67928](https://github.com/meltano/sdk/commit/7b679287f41a7fc6f8260a4da5c88acc94842f96).

@ReubenFrankel was the metadata validation failing before that commit?

Comment (ReubenFrankel) [reply cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml:34]: I didn't check. I just felt like it would be fine to always include given that a user may end up adding a `LICENSE` file later on.

**Comment by ReubenFrankel** [comment]
> > This AI review of the PR is crazy 😮
> 
> Crazy as "ok, this is actually somewhat useful" or "what? that is nonsense!"? 😅

Crazy as in I am impressed at how it summarised the changes and generated a whole flowchart too! Undecided on whether it is beneficial for the author or not, by can see it being useful to maintainers. I wonder how it would perform with bigger PRs...

File: cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml (hunk 1 above)
Comment (edgarrmondragon) [reply cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml:34]: That makes sense 👍

**Comment by edgarrmondragon** [comment]
> Crazy as in I am impressed at how it summarised the changes and generated a whole flowchart too! Undecided on whether it is beneficial for the author or not, by can see it being useful to maintainers. I wonder how it would perform with bigger PRs...

The summary has helped me a bit to catch things in a PR that shouldn't have been included, though I've yet to try it in larger PR.

**Comment by edgarrmondragon** [comment]
@ReubenFrankel what do you think adding a replay file in `e2e-tests/cookiecutters` to generate a project with license `None`. We can download the files and inspect them manually.

**Comment by edgarrmondragon** [comment]
> @ReubenFrankel what do you think adding a replay file in `e2e-tests/cookiecutters` to generate a project with license `None`. We can download the files and inspect them manually.

4179f392

**Comment by edgarrmondragon** [comment]
> > @ReubenFrankel what do you think adding a replay file in `e2e-tests/cookiecutters` to generate a project with license `None`. We can download the files and inspect them manually.
> 
> [4179f39](https://github.com/meltano/sdk/commit/4179f3921463b76107e145483dad7ec460f3f54a)

And here's how it looks

<details><summary><code>pyproject.toml</code></summary>

```toml
[project]
name = "tap-no-license"
version = "0.0.1"
description = "Singer tap for AutomaticTestTap, built with the Meltano Singer SDK."
readme = "README.md"
authors = [{ name = "Automatic Tester", email = "auto.tester@example.com" }]
keywords = [
    "ELT",
    "AutomaticTestTap",
]
classifiers = [
    "Intended Audience :: Developers",
    "Operating System :: OS Independent",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
    "Programming Language :: Python :: 3.13",
]
license-files = [ "LICENSE" ]
requires-python = ">=3.9"
dependencies = [
    "singer-sdk~=0.44.1",
    "requests~=2.32.3",
]

[project.optional-dependencies]
s3 = [
    "fs-s3fs~=1.1.1",
]

[project.scripts]
# CLI declaration
tap-no-license = 'tap_no_license.tap:TapAutomaticTestTap.cli'

[dependency-groups]
dev = [
    "pytest>=8",
    "singer-sdk[testing]~=0.44.1",
]

[tool.pytest.ini_options]
addopts = [
    "--durations=10",
]

[tool.mypy]
python_version = "3.12"
warn_unused_configs = true

[tool.ruff]
target-version = "py39"

[tool.ruff.lint]
ignore = [
    "COM812",  # missing-trailing-comma
]
select = ["ALL"]

[tool.ruff.lint.flake8-annotations]
allow-star-arg-any = true

[tool.ruff.lint.pydocstyle]
convention = "google"

[build-system]
requires = [
    "hatchling>=1,<2",
]
build-backend = "hatchling.build"
```

<p>


</p>
</details> 

And it is valid according to https://github.com/abravalheri/validate-pyproject.

**Review by edgarrmondragon — APPROVED** [review]
Thanks @ReubenFrankel!

**Comment by ReubenFrankel** [comment]
@edgarrmondragon Sorry - meant to say I did test this locally. Wasn't sure how the template tests worked so thanks for sorting. 👍
